### PR TITLE
Add dataset address to compute env variables

### PR DESCRIPTION
**Pull request by zguesmi**
* Add input files support in TEE workflow.
* A deal's dataset is considered valid only if all attributes are set (address, url, name, checksum).
* One env variable changes from `IEXEC_NB_INPUT_FILES` to `IEXEC_INPUT_FILES_NUMBER`.
* Move logback file to test package.

### CODE REVIEW

File: src/main/java/com/iexec/common/task/TaskDescription.java [hunk 1]
```diff
@@ -40,42 +41,93 @@
...
+                !StringUtils.isEmpty(datasetName);
+    }
+
+    /**
+     * Check if a callback is requested for this task. 
+     * 
+     * @return true if a callback address is found in the deal, false otherwise.
+     */
+    public boolean containsCallback() {
+        return getCallback() != null &&
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/task/TaskDescription.java]: !StringUtils.isEmpty && !equals(EMPTY_ADDRESS) ?

File: src/main/java/com/iexec/common/task/TaskDescription.java [hunk 2]
```diff
@@ -40,42 +41,93 @@
...
+     * non-null values. The stack should ignore datasets with missing
+     * information since they, inevitably, break the workflow. In the case
+     * where those datasets are ignored, the worker will contribute an
+     * application error caused by the missing dataset file.
+     * 
+     * @return true if all dataset fields are all non-null,
+     * non-empty values, false otherwise.
+     */
+    public boolean containsDataset() {
+        return !StringUtils.isEmpty(datasetAddress) &&
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/task/TaskDescription.java:75]: No need to add  !equals(EMPTY_ADDRESS) ?

File: src/main/java/com/iexec/common/utils/IexecEnvUtils.java [hunk 3]
```diff
@@ -33,38 +33,84 @@
...
+    /**
+     * Get compute stage environment variables plus other additional ones
+     * used by the pre-compute stage (e.g. IEXEC_DATASET_URL,
+     * IEXEC_DATASET_CHECKSUM, IEXEC_INPUT_FILE_URL_1, ...etc).
+     * 
+     * @param taskDescription
+     * @return
+     */
+    public static Map<String, String> getAllIexecEnv(TaskDescription taskDescription) {
+        Map<String, String> map = new HashMap<>();
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/utils/IexecEnvUtils.java]: Minor: "map" -> "envMap" (?)

File: src/main/java/com/iexec/common/utils/IexecEnvUtils.java [hunk 4]
```diff
@@ -33,38 +33,84 @@
...
+    public static final String IEXEC_DATASET_ADDRESS = "IEXEC_DATASET_ADDRESS";
+    public static final String IEXEC_DATASET_URL = "IEXEC_DATASET_URL";
+    public static final String IEXEC_DATASET_FILENAME = "IEXEC_DATASET_FILENAME";
+    public static final String IEXEC_DATASET_CHECKSUM = "IEXEC_DATASET_CHECKSUM";
+    // BoT
+    public static final String IEXEC_BOT_TASK_INDEX = "IEXEC_BOT_TASK_INDEX";
+    public static final String IEXEC_BOT_SIZE = "IEXEC_BOT_SIZE";
+    public static final String IEXEC_BOT_FIRST_INDEX = "IEXEC_BOT_FIRST_INDEX";
+    // input files
+    public static final String IEXEC_INPUT_FILES_NUMBER = "IEXEC_INPUT_FILES_NUMBER";
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/utils/IexecEnvUtils.java:49]: :+1:

**Review by james-toussaint — APPROVED** [review]

File: src/main/java/com/iexec/common/task/TaskDescription.java (hunk 1 above)
Comment (zguesmi) [reply src/main/java/com/iexec/common/task/TaskDescription.java]: Updated, thanks!

File: src/main/java/com/iexec/common/task/TaskDescription.java (hunk 2 above)
Comment (zguesmi) [reply src/main/java/com/iexec/common/task/TaskDescription.java:75]: It is already handled [here](https://github.com/iExecBlockchainComputing/iexec-common/pull/238/files#diff-97c4da1cbc899dfa86abb540221a1acaad5f012e2454c0618fc6542a0171698aR68). But I'll add here also.

**Pull request merged**